content/courses/tokens-and-nfts/nfts-with-metaplex-core.md

```diff
@@ -0,0 +1,834 @@
+with it. While this doesn't have to be an image, most Assets have an image associated 
+with them.
+
+Preparing and uploading an image involves converting to a buffer first. You can convert
+the file to a [generic file](https://developers.metaplex.com/umi/storage#generic-files) 
+using the `createGenericFile()` function and finally uploading it to the designated 
+Storage Driver.
+
+The `GenericFile` type allows Umi to support different file variations despite the 
+difference of browser files and local file system files i.e. those on your computer.
+
+In action, uploading an image named `random-image.png` from your computer would take the following steps:
+
+1. Reading the file using `readFile` into a buffer.
+
+2. Creating a generic file type with the files MIME Type from the buffer and filePath.
+
+3. Uploading file to designated storage provider.
+
+```typescript
+let filePath = "random-image.png";
+
+const buffer = await fs.readFile(filePath);
+let file = createGenericFile(buffer, filePath, {
+  // chose the correct file MIME type https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types
+  contentType: "image/jpeg",
+});
+const [image] = await umi.uploader.upload([file]);
+```
+
+The function will return the uri where the image is stored.
+
+#### Uploading the metadata
+
+After uploading the image, it's time to upload the offchain JSON metadata using the 
+`uploadJson()` method. This will return a uri where the JSON metadata is stored.
+
+Remember, the offchain portion of the metadata includes things like the image uri we
+just generated as well as additional information like the name and description of the 
+Asset. While you can technically include anything you'd like in this JSON object, in 
+most cases, you should follow the [NFT standard](https://developers.metaplex.com/token-metadata/token-standard#the-non-fungible-standard)
+to ensure compatibility with wallets, programs, and applications.
+
+To create the metadata, use the `uploadJson()` method provided by the SDK. This method 
+accepts a metadata object and returns a uri that points to the uploaded metadata.
+
+```typescript
+const metadata = {
+  name: 'My NFT',
+  description: 'This is an NFT on Solana',
+  image,
+  external_url: 'https://example.com',
+  properties: {
+    files: [
+      {
+        uri: imageUri[0],
+        type: 'image/jpeg',
+      },
+    ],
+    category: 'image',
+  },
+}
+
+const uri = await umi.uploader.uploadJson(metadata);
+```
+
+#### Creating the Collection
+
+Once the metadata is uploaded, you can finally create a Collection to group assets. 
+
+The `createCollection` method allows you to create a collection with the provided data
+
+```typescript
+const collection = generateSigner(umi)
+console.log("\nCollection Address: ", collection.publicKey.toString())
+
+const { signature, result } = await createCollection(umi, {
+  collection,
+  name: 'My Collection',
+  uri: collectionUri,
```

Comment:
Haven't described how/where to upload or where the `collectionuri` variable has come from.